### PR TITLE
fix dataset project settings

### PR DESCRIPTION
**Pull request by seanzhougoogle**
the project set in the bigquery client should be the default project to list dataset and run BQ jobs

so the service account specified should have "BigQuery Data Viewer" role on the "dataset project" and "BigQuery Job User" role on the "project"

### CODE REVIEW

File: src/test/java/io/cdap/plugin/gcp/bigquery/connector/BigQueryConnectorTest.java [hunk 1]
```diff
@@ -177,7 +188,8 @@ private void testBrowse(BigQueryConnector connector) throws IOException {
     Assert.assertEquals(1, detail.getTotalCount());
     Assert.assertEquals(1, detail.getEntities().size());
     for (BrowseEntity entity : detail.getEntities()) {
-      Assert.assertEquals("TABLE", entity.getType());
+      System.out.println(entity.getType() + " : " + entity.getName());
```
Comment (yaojiefeng) [inline src/test/java/io/cdap/plugin/gcp/bigquery/connector/BigQueryConnectorTest.java:201]: nit - remove this

File: src/test/java/io/cdap/plugin/gcp/bigquery/connector/BigQueryConnectorTest.java [hunk 2]
```diff
@@ -167,7 +177,8 @@ private void testBrowse(BigQueryConnector connector) throws IOException {
     Assert.assertTrue(detail.getTotalCount() > 0);
     Assert.assertTrue(detail.getEntities().size() > 0);
     for (BrowseEntity entity : detail.getEntities()) {
-      Assert.assertEquals("TABLE", entity.getType());
+      System.out.println(entity.getType() + " : " + entity.getName());
```
Comment (yaojiefeng) [inline src/test/java/io/cdap/plugin/gcp/bigquery/connector/BigQueryConnectorTest.java:189]: nit - remove this

File: src/test/java/io/cdap/plugin/gcp/bigquery/connector/BigQueryConnectorTest.java [hunk 3]
```diff
@@ -157,6 +166,7 @@ private void testBrowse(BigQueryConnector connector) throws IOException {
     Assert.assertTrue(detail.getTotalCount() > 0);
     Assert.assertTrue(detail.getEntities().size() > 0);
     for (BrowseEntity entity : detail.getEntities()) {
+      System.out.println(entity.getType() + " : " + entity.getName());
```
Comment (yaojiefeng) [inline src/test/java/io/cdap/plugin/gcp/bigquery/connector/BigQueryConnectorTest.java:177]: nit - remove this

File: src/main/java/io/cdap/plugin/gcp/bigquery/connector/BigQueryConnector.java [hunk 4]
```diff
@@ -129,39 +130,39 @@ public void test(FailureCollector failureCollector) throws ValidationException {
   public BrowseDetail browse(BrowseRequest browseRequest) throws IOException {
     BigQueryPath path = new BigQueryPath(browseRequest.getPath());
 
-    BigQuery bigQuery = getBigQuery();
-    if (path.isRoot()) {
+    String dataset = path.getDataset();
+    if (dataset == null) {
       // browse project to list all datasets
-      return listDatasets(bigQuery, browseRequest.getLimit());
+      return listDatasets(getBigQuery(config.getDatasetProject()), browseRequest.getLimit());
```
Comment (yaojiefeng) [inline src/main/java/io/cdap/plugin/gcp/bigquery/connector/BigQueryConnector.java:138]: A little confused here, there are couple places `getBigQuery` is called, sometimes it uses project, sometimes it uses dataset project, what is the difference and how do we decide which project to use?

**Review by yaojiefeng — COMMENTED** [review]
Some comments

File: src/main/java/io/cdap/plugin/gcp/bigquery/connector/BigQueryConnector.java (hunk 4 above)
Comment (seanzhougoogle) [reply src/main/java/io/cdap/plugin/gcp/bigquery/connector/BigQueryConnector.java:138]: as commented in getBigQuery method, it's where BQ list dataset and where the BQ job is run. 

So to list dataset, we should use `dataset project`. List tables already have dataset set in the tableId, so for other call we just use `project`

File: src/test/java/io/cdap/plugin/gcp/bigquery/connector/BigQueryConnectorTest.java (hunk 3 above)
Comment (seanzhougoogle) [reply src/test/java/io/cdap/plugin/gcp/bigquery/connector/BigQueryConnectorTest.java:177]: done

File: src/test/java/io/cdap/plugin/gcp/bigquery/connector/BigQueryConnectorTest.java (hunk 2 above)
Comment (seanzhougoogle) [reply src/test/java/io/cdap/plugin/gcp/bigquery/connector/BigQueryConnectorTest.java:189]: done

File: src/main/java/io/cdap/plugin/gcp/bigquery/connector/BigQueryConnector.java (hunk 4 above)
Comment (seanzhougoogle) [reply src/main/java/io/cdap/plugin/gcp/bigquery/connector/BigQueryConnector.java:138]: also add a javadoc for that method to make it more clear

**Review by yaojiefeng — APPROVED** [review]

**Pull request merged**